thoth/storages/security_indicators.py

```diff
@@ -59,45 +103,63 @@ def si_aggregated_document_exists(self) -> bool:
         """Check if the there is an object with the given key in bucket."""
         return self.ceph.document_exists("aggregated")
 
-    def connect(self) -> None:
-        """Connect this adapter."""
-        self.ceph.connect()
-
-    def is_connected(self) -> bool:
-        """Check if this adapter is connected."""
-        return self.ceph.is_connected()
-
-    def check_connection(self):
-        """Check connections of this adapter."""
-        self.ceph.check_connection()
-
 
 class SecurityIndicatorsResultsStore:
-    """An adapter for working with security indicators results."""
+    """Adapter for manipulating with Security Indicators."""
 
-    def __init__(self) -> None:
-        """Constructor."""
-        bucket_prefix = os.environ["THOTH_CEPH_BUCKET_PREFIX"]
-        deployment_name = os.environ["THOTH_DEPLOYMENT_NAME"]
-        prefix = f'{bucket_prefix}/{deployment_name}/security-indicators'
-        self.ceph = CephStore(prefix=prefix)
+    __slots__ = ["bandit", "cloc", "aggregated", "security_indicator_id"]
+
+    def __init__(self, security_indicator_id: str) -> None:
+        """A representation of a Security Indicator."""
+        self.security_indicator_id = security_indicator_id
+        self.bandit = SIBanditStore(security_indicator_id)
+        self.cloc = SIClocStore(security_indicator_id)
+        self.aggregated = SIAggregatedStore(security_indicator_id)
 
     def connect(self) -> None:
         """Connect this adapter."""
-        self.ceph.connect()
+        self.bandit.connect()
+        self.cloc.connect()
+        self.aggregated.connect()
 
     def is_connected(self) -> bool:
         """Check if this adapter is connected."""
-        return self.ceph.is_connected()
+        return self.bandit.is_connected() and self.cloc.is_connected() and self.aggregated.is_connected()
 
     def check_connection(self):
         """Check connections of this adapter."""
-        self.ceph.check_connection()
-
-    def get_listing(self) -> Generator[str, None, None]:
-        """Get listing of documents available in Ceph as a generator."""
-        return self.ceph.get_document_listing()
-
-    def get_document_count(self) -> int:
-        """Get number of documents present."""
-        return len(tuple(self.get_listing()))/3
+        self.bandit.check_connection()
+        self.cloc.check_connection()
+        self.aggregated.check_connection()
+
+    def exists(self, si_type: str) -> bool:
```

Comment:
Not sure about usage of this, it can be substituted with a more type-safe version:

```python
store = SecurityIndicatorsResultsStore("some-id")
store.connect()
store.si_aggregated_document_exists()
```

for the other two (bandit, cloc) respective methods could be introduced in used adapters.

Reply:
... and maybe we should completely discard name of the target from the method so it gets abstracted away. That way we can use adapters without any notion which results they encapsulate (and inherit from a common interface).